### PR TITLE
fix: Dask zero division error if parquet dataset has only one partition

### PR DESCRIPTION
**Pull request by mzwiessele**
Signed-off-by: Max Zwiessele <ibinbei@gmail.com>

**What this PR does / why we need it**:
When loading data in parquet dataset format it can happen that the dataset only has one partition in the `event_timestamp` column. If that is the case, dask will fail to process the dataset, erroring with a `ZeroDividionError` similar to 

https://github.com/feast-dev/feast/blob/769c31869eb8d9bb693f8a2876cc68b8cdd16521/sdk/python/feast/infra/offline_stores/file.py#L327

This PR adds a `try-catch` block to gracefully circumvent the error and process the data in only one partition.

**Which issue(s) this PR fixes**:

N/A

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/file.py [hunk 1]
```diff
@@ -662,14 +662,33 @@ def _drop_duplicates(
     created_timestamp_column: str,
     entity_df_event_timestamp_col: str,
 ) -> dd.DataFrame:
-    if created_timestamp_column:
-        df_to_join = df_to_join.sort_values(
-            by=created_timestamp_column, na_position="first"
-        )
+    column_order = df_to_join.columns
```
Comment (mzwiessele) [inline sdk/python/feast/infra/offline_stores/file.py:665]: Needed as during `ZeroDivisionError` the column order changes.

**Review by achals — APPROVED** [review]
Could you add a test for this case?

**Comment by mzwiessele** [comment]
> Could you add a test for this case?
Yes, I need help though. We need a test folder with a parquet dataset in the test S3 bucket. That parquet dataset must have only one partition in the `event_timestamp` column. The reason is simply that this code correction fixes exactly that edge case.

**Comment by mzwiessele** [comment]
Before merging this: Is it possible to update the Dask version feast relies on? Or in other words, why is the version restricted like this?
```
"dask>=2021.*,<2022.02.0"
```

**Comment by mzwiessele** [comment]
@achals I'll need help with the tests. Happy to do the ground work. Please point me to the right testing suite (in the unit tests) for loading a local parquet file from a FileSource.

**Comment by mzwiessele** [comment]
@achals I've added the test, please take a look and tell me if this does the trick!

**Comment by felixwang9817** [comment]
hey @mzwiessele thanks for the PR! a few responses to your comments above:

> Yes, I need help though. We need a test folder with a parquet dataset in the test S3 bucket. That parquet dataset must have only one partition in the event_timestamp column. The reason is simply that this code correction fixes exactly that edge case.

unless I'm misunderstanding, there's no need for the parquet dataset to be in an S3 bucket, right? the only thing that matters is that the Parquet dataset doesn't have `npartitions` set

if that's correct, I think the best way to write a test would be to do it locally - you can check out our unit tests in `sdk/python/feast/infra/tests/unit`, all of which run locally. I would try to imitate `test_e2e_local` in `test_e2e_local.py` - it's a good example of setting up a dataset locally, constructing a feature repo + `FeatureStore` object, and then running some tests, in your case I think you would just want to check that `get_historical_features` works on a Parquet dataset with `npartitions` unset

> Before merging this: Is it possible to update the Dask version feast relies on? Or in other words, why is the version restricted like this?

I forget exactly why we restricted that version; I'll go back and check, and if there's no strong reason I'm happy to bump up the upper bound restriction (although I think that can happen in a follow up PR)

**Comment by felixwang9817** [comment]
@mzwiessele also left some additional comments for you on #3235!

**Comment by mzwiessele** [comment]
@felixwang9817 I've added the dataset source to the test suite as suggested at #3235 to test loading of parquet datasets. 

Do you know if any of the created datasets feature only one partition in the event_timestamp column?

**Comment by mzwiessele** [comment]
kind ping :)

**Comment by achals** [comment]
> @felixwang9817 @achals kind ping :)

The DCO check still seems to be failing! Good to merge as soon as that's fixed. https://github.com/feast-dev/feast/pull/3236/checks?check_run_id=8744237672

**Comment by mzwiessele** [comment]
@achals fixed :)

**Review by achals — APPROVED** [review]
/lgtm
